process-manager.js

```diff
@@ -42,19 +42,18 @@ class ProcessWrapper extends EventEmitter {
 	}
 }
 
+// execFile - the path to the file to spawn the child process(es) from
+// maxProcesses - the maximum number of child processes to spawn
+// isChatBased - the process manager handles some chat functionality
 class ProcessManager {
 	constructor(options) {
-		if (!options) options = {};
-
 		this.processes = [];
 		this.taskId = 0;
+		this.execFile = options.execFile;
+		this.maxProcesses = options.maxProcesses;
```

Comment:
I suppose, but I was assuming the object would keep the same shape for any subclass constructed. Would a warning to `Monitor.log` be reasonable if a key needed is missing, and have valid values for backup?